tools/src/keygen/KeygenOptions.hs

```diff
@@ -43,21 +42,103 @@ secretDataTypeParser = asum
+        help "Password from the key passed in input."
+
+viewParser :: Parser (Maybe View)
+viewParser = optional $ asum
+    [ flag' RawView $
+          long "raw" <>
+          help "Display output as is"
+    , flag' HexView $
+          long "hex" <>
+          help "Display output in hex"
+    , flag' HexView $
+          long "base16" <>
+          help "Display output in hex"
+    , flag' Base64View $
+          long "base64" <>
+          help "Display output in base64"
+    ]
+
+prettinessParser :: Parser (Maybe Pretty)
+prettinessParser = optional $ asum
+    [ flag' (Pretty True) $
+          long "pretty" <>
+          help "Format output in pretty way"
+    , flag' (Pretty False) $
+          long "one-line" <>
+          help "Print output in one line"
+    ]
 
 keygenCommandParser :: Parser KeygenCommand
-keygenCommandParser = option keygenCommandReadM $
-    long "command" <>
-    metavar "TEXT" <>
-    help "Which output should be produced. See README.md for details"
+keygenCommandParser = hsubparser $ mconcat
+    [ command "secret" $ info
+        (PrintSecretKey . fromMaybe Base64View <$> viewParser)
+        (fullDesc <>
+         progDesc "Display secret itself. In base64 by default.")
+
+    , command "public" $ info
+        (PrintPublicKey . fromMaybe HexView <$> viewParser)
+        (fullDesc <>
+         progDesc "Display the public key corresponding to the secret. In hex by default.")
+
+    , command "address" $ info
+        (pure PrintAddress)
+        (fullDesc <>
+         progDesc "Display the address derivative from the secret.")
+
+    , command "esecret" $ info
+        (PrintEncryptedSecretKey <$> passphraseParser <*> (viewParser <?:> Base64View))
+        (fullDesc <>
+         noIntersperse <>
+         -- needed since --password may be both an option or a command argument otherwise
+         progDesc "Display secret key encrypted. Note that if no password is given, an \
+                  \encrypted secret will be produced anyway (which is ~twice as long as \
+                  \an unencrypted secret.")
+
+    , command "keyfile" $ info
+        (PrintKeyFile <$> passphraseParser <*> (prettinessParser <?:> Pretty True))
+        (fullDesc <>
+         noIntersperse <>
+         progDesc "Display keyfile content. You will probably want to redirect the output \
+                  \of this command to the file.")
+
+    , command "educator-auth" $ info
+        (fmap PrintEducatorAuthToken . strOption $
+            long "endpoint" <>
+            metavar "TEXT" <>
+            help "Exact name of signed endpoint.")
+        (fullDesc <>
+         progDesc "Create a JWT signature for authentication in Educator/Student API. \
+                  \You have to provide the name of signed endpoint, for instance \
+                  \`/api/student/v1/submissions`")
+
+    , command "student-submission" $ info
+        (fmap PrintStudentSubmission . strOption $
+            long "seed" <>
```

Comment:
Is there any other way to solve this?

Reply:
I extracted that parser and one below, if you were talking about it.